x-pack/plugins/fleet/server/services/agents/uploads.ts

```diff
@@ -216,6 +231,61 @@ export async function getAgentUploadFile(
   }
 }
 
+export async function deleteAgentUploadFile(
```

Comment:
There aren't unit tests right now for this module, but it would probably be good to add them just for this function in this PR. We'll probably want to cover this function + the `DELETED` check above in the `getAgentUploads` method. We can file a follow-up to flesh out the rest of the test coverage since it wasn't added initially. 

I know there's FTR coverage included here, but I think we should add direct unit tests where possible too.

Reply:
unit tests added for these two methods!

instead of testing against the `DELETED` status (since these items are _not_ supposed to be returned from `getAgentUploads`), I added a comprehensive fixture to test against various states of agent action + action results + file metadata and ensured that the method returns the right list of items